project/SparkBuild.scala

```diff
@@ -790,7 +790,7 @@ object TestSettings {
     javaOptions in Test ++= System.getProperties.asScala.filter(_._1.startsWith("spark"))
       .map { case (k,v) => s"-D$k=$v" }.toSeq,
     javaOptions in Test += "-ea",
-    javaOptions in Test ++= "-Xmx3g -Xss4096k"
+    javaOptions in Test ++= "-Xmx6g -Xss4096k"
```

Comment:
I'm a little worried about this change. Since the change to `BlockManagerSuite` is not very related to this PR, can we revert and revisit it in follow-up PR? Then we can unblock this PR.